gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ObservationCommentMapperTest.java

```diff
@@ -0,0 +1,141 @@
+        assertThat(observation.getMeta().getProfile().get(0).getValue()).isEqualTo(META_URL);
+        assertThat(observation.getStatus()).isEqualTo(Observation.ObservationStatus.FINAL);
+        assertThat(observation.getSubject().getResource()).isEqualTo(patient);
+
+        var coding = observation.getCode().getCoding();
+        assertThat(coding.get(0).getCode()).isEqualTo(CODING_CODE);
+        assertThat(coding.get(0).getSystem()).isEqualTo(CODING_SYSTEM);
+
+        assertThat(observation.getEffective().toString()).isEqualTo(
+            DateFormatUtil.parseToDateTimeType(narrativeStatement.getAvailabilityTime().getValue()).toString());
+
+        assertThat(observation.getIssuedElement().asStringValue()).isEqualTo("2010-01-14T00:00:00.000+00:00");
+
+        var identifier = observation.getIdentifier().get(0);
+        assertThat(identifier.getValue()).isEqualTo(narrativeStatementId);
+        assertThat(identifier.getSystem()).isEqualTo(IDENTIFIER_SYSTEM);
+
+        assertThat(observation.getPerformer().get(0).getReference()).isEqualTo("Practitioner/2D70F602-6BB1-47E0-B2EC-39912A59787D");
+        assertThat(observation.getComment()).isEqualTo("Some example text");
+        assertThat(observation.getContext().getResource()).isEqualTo(encounter);
+    }
+
+    @Test
+    public void mapObservationsWithFullDataMultipleObservations() {
+        var ehrExtract = unmarshallCodeElement("multiple_narrative_statements.xml");
+
+        List<Observation> observations =
+            observationCommentMapper.mapObservations(ehrExtract, patient, Collections.emptyList());
+
+        assertThat(observations.size()).isEqualTo(EXPECTED_OBSERVATION_COUNT);
+    }
+
+    @Test
+    public void mapObservationsNoObservationToMap() {
+        var ehrExtract = unmarshallCodeElement("no_narrative_statement.xml");
+
+        List<Observation> observations =
+            observationCommentMapper.mapObservations(ehrExtract, patient, Collections.emptyList());
+
+        assertThat(observations).isEmpty();
+    }
+
+    @Test
+    public void mapObservationsCompositionHasNoAuthorTime() {
+        var ehrExtract = unmarshallCodeElement("nullflavour_composition_author_time.xml");
+
+        List<Observation> observations =
+            observationCommentMapper.mapObservations(ehrExtract, patient, Collections.emptyList());
+
+        // Value taken from the extract's availability time
+        assertThat(observations.get(0).getIssuedElement().asStringValue()).isEqualTo("2020-01-01T01:01:01.000+00:00");
+    }
+
+    @Test
+    public void mapObservationWithFullDataNoMappedEncounter() {
+        var ehrExtract = unmarshallCodeElement("single_narrative_statement.xml");
+
+        List<Observation> observations =
+            observationCommentMapper.mapObservations(ehrExtract, patient, Collections.emptyList());
+
+        // Calling `getContext` auto creates a Reference object so asserting the reference is null
```

Comment:
I've seen comments used throughout the PR, are they entirely necessary? I guess its up for debate, although my argument would be that comments (at present) add noise and are usually only used in this project to highlight important implementations yet to be developed (TODOs). In the case of these tests, maybe rename the tests themselves to note what the comment says?

Reply:
As discussed privately, we've ditched one and left the other two